Framework/Kernel/src/TimeSeriesProperty.cpp

```diff
@@ -1411,15 +1426,21 @@ void TimeSeriesProperty<TYPE>::create(const std::vector<Types::Core::DateAndTime
                                 "for the time and values vectors.");
 
   clear();
-  m_values.reserve(new_times.size());
+  // nothing to do without values
+  if (new_times.empty()) {
+    m_propSortedFlag = TimeSeriesSortStatus::TSSORTED;
+    return;
+  }
 
-  std::size_t num = new_values.size();
+  const std::size_t num = new_values.size();
+  m_values.reserve(num);
 
   m_propSortedFlag = TimeSeriesSortStatus::TSSORTED;
-  for (std::size_t i = 0; i < num; i++) {
-    TimeValueUnit<TYPE> newentry(new_times[i], new_values[i]);
-    m_values.emplace_back(newentry);
-    if (m_propSortedFlag == TimeSeriesSortStatus::TSSORTED && i > 0 && new_times[i - 1] > new_times[i]) {
+  // add the 0th value to simplify logic inside of the loop
```

Comment:
I would split this into two:
```cpp
m_propSortedFlag = std::is_sorted(new_times) ? TimeSeriesSortStatus::TSSORTED : TimeSeriesSortStatus::TSUNSORTED;
for (std::size_t 0 = 1; i < num; i++)
    m_values.emplace_back(new_times[i], new_values[i]);
```
The reasons is that once you find the first violating `new_times[i - 1] > new_times[i]`, then there's no point in checking other `i-1, i` pairs.

Reply:
I wasn't sure if `std:::sort` would be faster or if checking the order while looping through. The `if` on L1443 is checking that the flag is sorted first which should be cheap (enum comparison). I tried it out and we get another second out of the python script in the PR description.